### PR TITLE
React 16

### PR DESCRIPTION
**Pull request by tmkelly28**
* Update to React v16
* Needed to update enzyme as well
* Needed to update the test script, because otherwise it was unnecessarily grabbing some tests from deeply within the node modules folder

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "start": "node server",
     "start-dev": "npm run build-client-watch & npm run start-server",
     "start-server": "NODE_ENV='development' nodemon server -e html,js,scss --ignore public",
-    "test": "NODE_ENV='test' DATABASE_URL='postgres://localhost:5432/boilermaker-test' mocha ./**/*.spec.js --compilers js:babel-register"
+    "test": "NODE_ENV='test' DATABASE_URL='postgres://localhost:5432/boilermaker-test' mocha ./server/**/*.spec.js ./server/**/**/*.spec.js ./client/**/*.spec.js --compilers js:babel-register"
```
Comment (glebec) [inline package.json]: I thought `**` in a shell glob meant recursively to go down any number of directories. Why both `./server/**/*.spec.js` and `./server/**/**/*.spec.js`?

File: .babelrc [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  presets: ['react', 'es2015', 'stage-0']
+  presets: ['react', 'env', 'stage-0']
```
Comment (glebec) [inline .babelrc]: I think Babel-Preset-Env usually requires or at least recommends a modicum of configuration… I don't recall all the defaults offhand.

Also I am personally not a fan of `stage-0` ever. I feel that `stage-2` is a [much better threshold](https://tc39.github.io/process-document/) for inclusion in a project.

> Stage 1 Spec Quality: none.

> Stage 2 Spec Quality: Draft: all **major** semantics, syntax and API are covered, but TODOs, placeholders and editorial issues are expected.

I could make an argument that even stage 3 or 4 are preferable, but I get the fun of using new features "early."

Comment (glebec) [reply .babelrc]: [Yep](https://github.com/babel/babel-preset-env/blob/master/README.md#babel-preset-env----):

> Without any configuration options, babel-preset-env behaves exactly the same as babel-preset-latest

You can see an example config that works for Bones, with associated conversation, in [this PR](https://github.com/FullstackAcademy/bones/pull/106/files).

File: package.json (hunk 1 above)
Comment (tmkelly28) [reply package.json]: That's what I thought too, but unfortunately it doesn't seem to be doing that :/

**Comment by tmkelly28** [comment]
* Bumped the preset up to stage 2
* Added a blurb about what the presets are for

**Pull request merged**